embedded/object/engine_test.go

```diff
@@ -0,0 +1,76 @@
+package object
+
+import (
+	"context"
+	"testing"
+
+	"github.com/codenotary/immudb/embedded/sql"
+	"github.com/codenotary/immudb/embedded/store"
+	"github.com/stretchr/testify/require"
+)
+
+func closeStore(t *testing.T, st *store.ImmuStore) {
+	err := st.Close()
+	if !t.Failed() {
+		// Do not pollute error output if test has already failed
+		require.NoError(t, err)
+	}
+}
+
+func TestCreateCollection(t *testing.T) {
```

Comment:
test doesn't seem to be validating functionality

Reply:
Yes, I am adding more to it

Reply:
added functional tests, will make a separate PR to add more, have a list created